firefox_ui_tests/remote/security/test_green_larry.py

```diff
@@ -0,0 +1,109 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import expected
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestGreenLarry(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.url = 'https://ssl-ev.mozqa.com/'
+
+        self.identity_popup = self.browser.navbar.locationbar.identity_popup
+
+    def tearDown(self):
+        try:
+            # TODO: PR 105/115: insure identity_popup closes
+            self.windows.close_all([self.browser])
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_green_larry(self):
+        with self.marionette.using_context('content'):
+            self.marionette.navigate(self.url)
+
+        # Get the information from the certificate
+        cert = self.browser.tabbar.selected_tab.certificate
+        address = self.security.get_address_from_certificate(cert)
+
+        # Check the identity popup label displays
+        self.assertEqual(self.identity_popup.organization_label.get_attribute('value'),
+                         cert['organization'])
+        self.assertEqual(self.identity_popup.country_label.get_attribute('value'),
+                         '(' + address['country'] + ')')
+
+        # Check the favicon
+        # TODO: find a better way to check, e.g., mozmill's isDisplayed
+        favicon = self.browser.navbar.locationbar.favicon
+        self.wait_for_condition(lambda _: favicon.get_attribute('hidden') == 'false')
+
+        # Check the identity popup box
+        self.assertEqual(self.identity_popup.box.get_attribute('className'),
+                         'verifiedIdentity')
+
+        self.identity_popup.box.click()
+        self.wait_for_condition(expected.element_present(
+                                lambda _: self.identity_popup.popup))
```

Comment:
This doesn't look right to me. We are waiting for the popup to be open, and not that it exists. This wait will return immediately.